crates/once-map/src/lib.rs

```diff
@@ -46,41 +47,61 @@ impl<K: Eq + Hash, V: Clone, H: BuildHasher + Clone> OnceMap<K, V, H> {
     ///
     /// Will hang if [`OnceMap::done`] isn't called for this key.
     pub async fn wait(&self, key: &K) -> Option<V> {
-        let entry = self.items.get(key)?;
+        let notify = {
+            let entry = self.items.get(key)?;
+            match entry.value() {
+                Value::Filled(value) => return Some(value.clone()),
+                Value::Waiting(notify) => notify.clone(),
+            }
+        };
+
+        // Prepare to wait.
+        let mut notification = pin!(notify.notified());
+        notification.as_mut().enable();
+
+        // Make sure the value wasn't inserted in-between us checking the map and preparing to wait.
+        if let Value::Filled(value) = self.items.get(key).expect("map is append-only").value() {
+            return Some(value.clone());
+        };
+
+        // Wait until the value is inserted.
+        notification.await;
```

Comment:
OK, I buy this, because acquiring the `Notification` has been de-coupled from `await`ing on the `Notification`. And `notify_waiters` specifically says:

> The purpose of this method is to notify all already registered waiters. Registering for notification is done by acquiring an instance of the Notified future via calling notified().

But my reading here is still that `notification.as_mut().enable()` is not needed here. Not unless we're using `notify_{one,last}` somewhere.